### DISCUSSION TITLE
[SOLVED] Add support for mercury text parser

### DISCUSSION THREAD

**Discussion by user858753257** [martinrotter/rssguard#553] [Q&A]
### Brief description of the feature request

More infos here : 

https://mercury.postlight.com/

With a button a user can parse the rss content or open the browser

**Comment by martinrotter** [comment]
No need to really do anything. You can use post-process script feature to call Mercury without problems.

**Comment by user858753257** [comment]
> No need to really do anything. You can use post-process script feature to call Mercury without problems.

Where I can find this feature ?

**Comment by martinrotter** [comment]
Read documentation, I will write some summary here for you.

**Reply by user858753257** [reply]
Thanks  . I'm using the program on windows .

**Comment by martinrotter** [comment]
I converted issue into discussion.

**Comment by martinrotter** [comment]
So, I am not sure how exactly you want to use Mercury with RSS Guard but I guess you want to use it to have automatically fetched "mercuried" articles into your feed instead of original feed data?

**Reply by user858753257** [reply]
Either automatically or when pressing a button in the reading pane . 


What is the easiest way to read a full text ?

**Reply by martinrotter** [reply]
OK, so in that case, I made PoC for automatic fetching, which you can enable per-feed:
1. Download and install LATEST RSS Guard development [build](https://github.com/martinrotter/rssguard/releases/tag/devbuild). It contains some fixes which allow this use case to work.
2. Install Mercury:
    a. Install [NPM](https://www.npmjs.com/).
    b. Install Mercury as per project's instructions.
    c. Add `node.exe` and `npm.cmd` (or `npm` on non-Windows platforms) to your `PATH` environment variable, also setup `NODE_PATH` environment variable to point to directory where your modules (mercury-parser) are installed (not needed with recent dev. versions if packages are installed globally).
3. Once Mercury is installed and working, open RSS Guard and head to `Articles -> Article filters`.
4. Create new article filter with below contents:

```js
function filterMessage() {
  args = `const fs = require("@postlight/mercury-parser");fs.parse("${msg.url}").then(result => console.log(result.content));`;

  msg.contents = utils.runExecutableGetOutput("node.exe", ["-e", args]);

  return MessageObject.Accept;
}
```

This article filter is automatically called with newly downloaded articles and runs "node.exe" with URL of the article which downloads its "full" contents via `mercury-parser` and replaces existing contents of new article.

5. Check feeds you wish to have this filter applied to.
6. Update your feeds.
7. See some below screenshot(s).

### The dialog
![image](https://user-images.githubusercontent.com/1255302/146322037-51ae0274-5995-4060-b0db-618a686818e0.png)

### Sample of article automatically downloaded with `mercury-parser`
![image](https://user-images.githubusercontent.com/1255302/146322064-369f500f-1bef-483b-9b9d-fecee734a3b9.png)

Note that `mercury-parser` is not that "fast", it takes about 1.3 seconds to download/process one article, so fetching of one feed which could contain like 20 articles could take 30 seconds. Bigger feeds will take even more time.

**Reply by user858753257** [reply]
Perfekt . Thanks for the explanation .

The other two Programms I only need for the Dev Version of RssGuard ? 
Or also later when a new stable is published in future ?

**Reply by martinrotter** [reply]
Hi. No, you need those all the time. :)

**Reply by martinrotter** [reply]
Closing this as I provided some info and also, it seems that this is actually the same as https://github.com/martinrotter/rssguard/discussions/559 only with different "backend".
